modules/nominees-election/src/lib.rs

```diff
@@ -383,30 +426,40 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 	}
 
-	fn rebalance() {
+	fn rebalance() -> Result<(), Error<T, I>> {
 		let mut voters = Votes::<T, I>::iter().collect::<Vec<(T::NomineeId, Balance)>>();
 
 		voters.sort_by(|a, b| b.1.cmp(&a.1));
 
-		let new_nominees = voters
+		let new_nominees: BoundedVec<<T as Config<I>>::NomineeId, <T as Config<I>>::NominateesCount> = voters
 			.into_iter()
 			.take(T::NominateesCount::get().saturated_into())
 			.map(|(nominee, _)| nominee)
-			.collect::<Vec<_>>();
+			.collect::<Vec<_>>()
+			.try_into()
+			.map_err(|_| Error::<T, I>::NominateesCountExceeded)?;
```

Comment:
we can have `expect` here as the max numbers of elements is `NominateesCount` guaranteed by `take`